gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,383 @@
+ * from the solutions registry.
+ * @param {Object} solutions - Solutions registry entries for solutions available on the current machine.
+ * @param {Object} options - Extra options for processing.
+ * @param {Array} options.solutionsList - If provided, only solutions in this list of `solutionsID`s will
+ * be captures. Example:
+ *
+ * '''json
+ * ["com.microsoft.windows.cursors", "com.freedomscientific.jaws"]
+ * '''
+ * @return {fluid.promise} Returns a promise resolving with the entire system settings capture.
+ */
+gpii.flowManager.captureSystemSettings = function (invokeSettingsHandlerGet, resolver, localResolver, solutions, options) {
+    var promiseTogo = fluid.promise();
+
+    var handlerGetPromises = [];
+
+    var safeHandlerGet = function (invokeSettingsHandlerGet, solutionID, handlerSpec) {
+        var promiseTogo = fluid.promise();
+        try {
+            invokeSettingsHandlerGet(solutionID, handlerSpec).then(function (data) {
+                // Fix up the nativeWindows and SPI Settings Handlers
+                var filteredData = gpii.flowManager.filterInvokeSettingsHandlersReturn(data, handlerSpec);
+                promiseTogo.resolve(filteredData);
+            }, function (err) {
+                promiseTogo.resolve({
+                    solutionId: solutionID,
+                    handlerSpec: handlerSpec,
+                    isError: true,
+                    message: "inner promise error",
+                    err: err
+                });
+            });
+        }
+        catch (err) {
+            promiseTogo.resolve({
+                solutionId: solutionID,
+                handlerSpec: handlerSpec,
+                isError: true,
+                message: "outer try error",
+                err: err
+            });
+        }
+        return promiseTogo;
+    };
+
+    var solutionsToFetch = solutions;
+    if (options.solutionsList) {
+        solutionsToFetch = {};
+        fluid.each(solutions, function (solution, solutionID) {
+            if (options.solutionsList.indexOf(solutionID) >= 0) {
+                solutionsToFetch[solutionID] = solution;
+            }
+        });
+    }
+
+    fluid.each(solutionsToFetch, function (solution, solutionID) {
+        var newSolution = gpii.lifecycleManager.transformSolutionSettings(solution, gpii.settingsHandlers.changesToSettings);
+
+        if (solutionID === "com.freedomscientific.jaws") {
+            newSolution = gpii.flowManager.jawsVoiceProfileAdjustment(newSolution, localResolver);
+        }
+
+        fluid.each(newSolution.settingsHandlers, function (handlerSpec) {
+            //if applicationSettings is unspecified, dont attempt to get settings
+            if (!handlerSpec.supportedSettings) {
+                return;
+            }
+            handlerSpec.settings = handlerSpec.supportedSettings;
+            // TODO GPII-228 This is temporary debugging code while we work through
```

Comment:
Add the JIRA that describes the problem that this work around is here for.

Reply:
Forgot to mention, it might make more sense to move this handling for "gpii.windows.spiSettingsHandler" to gpii/windows repo too.

Reply:
This has been removed, and will work as soon as GPII-3119 https://github.com/GPII/universal/pull/832 is merged in.  Do note that without GPII-3119, SPI settings will likely cause the snapshotter to crash.